### PR TITLE
ceph-detect-init: return systemd on Debian Jessie

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/15086 Fixes: #15086

Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
Tested in a Jessie container. Without this patch, ceph-detect-init returns "sysvinit". With the patch, it returns "systemd".

File: src/ceph-detect-init/ceph_detect_init/debian/__init__.py [hunk 1]
```diff
@@ -14,4 +14,7 @@ def choose_init():
             return 'systemd'
         else:
             return 'upstart'
+    if distro.lower() in ('debian'):
+        if codename >= 'jessie':
```
Comment (liewegas) [inline src/ceph-detect-init/ceph_detect_init/debian/__init__.py]: the debian version names aren't sorted... i think we need to enumerate the "old" distros and map them to sysvinit, and then default to systemd for others.

i think that is just squeeze and wheezy -> sysvinit.. anything older won't work anyway.

**Comment by smithfarm** [comment]
@liewegas OK. By the way, the old distros are already enumerated - read on.

In the Jessie container, `codename` evaluates to an empty string:

```
>>> import platform
>>> print platform.linux_distribution()
('debian', '8.3', '')
```

The code takes this into account and enumerates here: https://github.com/ceph/ceph/blob/master/src/ceph-detect-init/ceph_detect_init/__init__.py#L87-L92

**Comment by smithfarm** [comment]
Tested on Jessie only so far.

**Comment by liewegas** [comment]
Thanks, looks good to me!

**Comment by smithfarm** [comment]
Should I retarget this to master?

**Comment by liewegas** [comment]
merged manually

**Pull request merged**